src/Nethermind/Nethermind.Core/Specs/ISpecProvider.cs

```diff
@@ -61,13 +62,13 @@ public interface ISpecProvider
         /// <summary>
         /// All block numbers at which a change in spec (a fork) happens.
         /// </summary>
-        long[] TransitionBlocks { get; }
+        ForkActivation[] TransitionBlocks { get; }
 
         /// <summary>
         /// Resolves a spec for the given block number.
         /// </summary>
         /// <param name="blockNumber"></param>
         /// <returns>A spec that is valid at the given chain height</returns>
-        IReleaseSpec GetSpec(long blockNumber);
+        IReleaseSpec GetSpec(ForkActivation forkActivation);
```

Comment:
lets also add overload or extension method that takes `IReleaseSpec GetSpec(int blockNumber, ulong? timestamp = 0);

Reply:
added it without the nullable timestamp, because i want it to only be used by the compiler when timestamp is provided, otherwise use the implicit operator to convert the blockNumber to forkActivation